habitat-baselines/habitat_baselines/rl/ppo/ppo_trainer.py

```diff
@@ -567,29 +452,34 @@ def _collect_rollout_step(self):
 
     @profiling_wrapper.RangeContext("_update_agent")
     def _update_agent(self):
-        ppo_cfg = self.config.habitat_baselines.rl.ppo
         t_update_model = time.time()
+
         with inference_mode():
-            step_batch = self.rollouts.buffers[
-                self.rollouts.current_rollout_step_idx
+            step_batch = self._agent.rollouts.buffers[
+                self._agent.rollouts.current_rollout_step_idx
             ]
 
-            next_value = self.actor_critic.get_value(
+            next_value = self._agent.actor_critic.get_value(
                 step_batch["observations"],
                 step_batch["recurrent_hidden_states"],
                 step_batch["prev_actions"],
                 step_batch["masks"],
             )
 
-        self.rollouts.compute_returns(
-            next_value, ppo_cfg.use_gae, ppo_cfg.gamma, ppo_cfg.tau
+        self._agent.rollouts.compute_returns(
+            next_value,
+            self._ppo_cfg.use_gae,
+            self._ppo_cfg.gamma,
+            self._ppo_cfg.tau,
         )
 
-        self.agent.train()
+        self._agent.train()
+
+        losses = self._agent.updater.update(self._agent.rollouts)
```

Comment:
I do not like calling directly rollouts and updater on _agent because it seems like there is a specific order these methods need to be called in. I would push this complexity down and create an `agent.update()` that would call 
```
self.rollouts.compute_returns(
            next_value,
            self._ppo_cfg.use_gae,
            self._ppo_cfg.gamma,
            self._ppo_cfg.tau,
        )
self.train()
self.updater.update(self._agent.rollouts)
self.rollouts.after_update()
self.after_update()
self.eval()
```
and maybe even the block inside with `inference_mode()`

If we do not want to do this (not sure why) we should at least isolate this logic into a static method that takes as input the agent and the _ppo_cfg paraeters.